src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt

```diff
@@ -107,34 +115,32 @@ class NexusPublishPlugin : Plugin<Project> {
 
     private fun configureTaskDependencies(
         project: Project,
-        initializeTask: TaskProvider<Task>,
-        closeTask: TaskProvider<Task>,
-        releaseTask: TaskProvider<Task>,
-        nexusRepository: MavenArtifactRepository
+        initializeTask: TaskProvider<InitializeNexusStagingRepository>,
+        publishAllTask: TaskProvider<Task>,
+        closeTask: TaskProvider<CloseNexusStagingRepository>,
+        releaseTask: TaskProvider<ReleaseNexusStagingRepository>,
+        mavenRepo: MavenArtifactRepository
     ) {
-        val publishTasks = project.tasks
-                .withType<PublishToMavenRepository>()
-                .matching { it.repository == nexusRepository }
-        project.tasks.register("publishTo${nexusRepository.name.capitalize()}") {
-            description = "Publishes all Maven publications produced by this project to the '${nexusRepository.name}' Nexus repository."
-            group = PublishingPlugin.PUBLISH_TASK_GROUP
-            dependsOn(publishTasks)
-        }
-        // PublishToMavenRepository tasks may not yet have been initialized
-        project.afterEvaluate {
-            publishTasks.configureEach {
+        val mavenPublications = project.the<PublishingExtension>().publications.withType<MavenPublication>()
+        mavenPublications.all {
+            val publication = this
+            val publishTask = project.tasks.withName<PublishToMavenRepository>(
+                    "publish${publication.name.capitalize()}PublicationTo${mavenRepo.name.capitalize()}Repository")
+            publishTask {
                 dependsOn(initializeTask)
                 doFirst { logger.info("Uploading to {}", repository.url) }
             }
-        }
-        closeTask {
-            mustRunAfter(initializeTask)
-            mustRunAfter(publishTasks)
-        }
-        releaseTask {
-            mustRunAfter(initializeTask)
-            mustRunAfter(closeTask)
-            mustRunAfter(publishTasks)
+            publishAllTask.configure {
+                dependsOn(publishTask)
+            }
+            closeTask {
+                mustRunAfter(publishTask)
+                mustRunAfter(publishAllTask)
+            }
+            releaseTask {
+                mustRunAfter(publishTask)
+                mustRunAfter(publishAllTask)
+            }
```

Comment:
Is it required to add `closeTask { mustRunAfter(publishTask) }` (and the same for `releaseTask`)?

I guess it should be enough to add `closeTask { mustRunAfter(publishAllTask) }` just once, not multiple times (for each publication)

Reply:
> Is it required to add `closeTask { mustRunAfter(publishTask) }` (and the same for `releaseTask`)?

Yes, since using the `publishAllTask` is not mandatory.

> I guess it should be enough to add `closeTask { mustRunAfter(publishAllTask) }` just once, not multiple times (for each publication)

Good catch!